core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -92,6 +94,36 @@ class KryoSerializer(conf: SparkConf)
       new KryoOutput(bufferSize, math.max(bufferSize, maxBufferSize))
     }
 
+  @transient
+  private lazy val factory: KryoFactory = new KryoFactory() {
+    override def create: Kryo = {
+      newKryo()
+    }
+  }
+
+  private class PoolWrapper extends KryoPool {
```

Comment:
This wrapper can be removed when Kryo is updated to 5.0 as the new pool implementation exposes the needed method to clear the pool